src/core/msg/format.rs

```diff
@@ -46,36 +46,36 @@ impl<'b> MessageFormatter<'b> {
         let mut f = MessageFormatter {
             buffer: buffer, cursor: 0, remaining_arguments: len,
         };
-        f.add_char(b'{')?;
-        f.encode(&len, len.get_size())?;
-        f.add_char(b'|')?;
-        f.add_argument(type_name)?;
-        Ok(f)
+        f.add_char(b'{');
+        f.encode(&len, len.get_size());
+        f.add_char(b'|');
+        f.add_argument(type_name);
+        f
     }
 
-    ///Convenience wrapper around `new()` and `finalize()` that allows you to
-    ///use the `?` operator on `add_argument()` even if your function's error
-    ///type does not match ours. The result value is the same as for
-    ///`finalize()`.
+    ///This convenience function uses MessageFormatter to format a `core.pub`
+    ///message into the given `buffer`. This is intended for usage by
+    ///implementations of
+    ///[`vt6::server::Handler::get_set_property()`](../../server/trait.Handler.html).
+    ///For example,
     ///
+    ///```rust,ignore
+    ///let result = MessageFormatter::publish_property(buf, "example.counter", &42);
     ///```
-    ///# use std;
-    ///# use vt6::core::msg::*;
-    ///let mut buf = vec![0u8; 1024];
-    ///let size = MessageFormatter::format(&mut buf, "have", 2, |f| {
-    ///    f.add_argument("core")?;
-    ///    f.add_argument("1.0") //TODO update example after adding ModuleVersion
-    ///}).unwrap();
     ///
-    ///let msg_str = std::str::from_utf8(&buf[0..size]).unwrap();
-    ///assert_eq!(msg_str, "{3|4:have,4:core,3:1.0,}");
+    ///is equivalent to:
+    ///
+    ///```rust,ignore
+    ///let mut f = MessageFormatter::new(buf, "core.pub", 2);
+    ///f.add_argument("example.counter");
+    ///f.add_argument(&42);
+    ///let result = f.finalize();
     ///```
-    pub fn format<T>(buffer: &'b mut [u8], type_name: &str, num_arguments: usize, callback: T) -> Result<usize, BufferTooSmallError>
-        where T: FnOnce(&mut MessageFormatter) -> Result<(), BufferTooSmallError>
-    {
-        let mut f = Self::new(buffer, type_name, num_arguments)?;
-        callback(&mut f)?;
-        f.finalize()
+    pub fn publish_property<T: EncodeArgument + ?Sized>(buffer: &'b mut [u8], property_name: &str, property_value: &T) -> Option<usize> {
```

Comment:
Open question: Leave this as it is right now, in `impl MessageFormatter`, or pull out into a separate module for pre-recorded messages, e.g. `vt6::core::msg::prerecorded::publish_property`?